website/server/libs/api-v3/taskManager.js

```diff
@@ -0,0 +1,155 @@
+import * as Tasks from '../../models/task';
+import {
+  BadRequest,
+} from '../api-v3/errors';
+import Bluebird from 'bluebird';
+import _ from 'lodash';
+
+async function _validateTaskAlias (tasks, res) {
+  let tasksWithAliases = tasks.filter(task => task.alias);
+  let aliases = tasksWithAliases.map(task => task.alias);
+
+  // Compares the short names in tasks against
+  // a Set, where values cannot repeat. If the
+  // lengths are different, some name was duplicated
+  if (aliases.length !== [...new Set(aliases)].length) {
+    throw new BadRequest(res.t('taskAliasAlreadyUsed'));
+  }
+
+  await Bluebird.map(tasksWithAliases, (task) => {
+    return task.validate();
```

Comment:
This is not strictly related to this PR but since we don't have async mongoose validators (as far as i remember) we could change this to use `task.validateSync()` which should be considerably faster as it's only sync code